ui/src/components/mobileservices/BoundServiceRow.js

```diff
@@ -91,13 +96,76 @@ class BoundServiceRow extends Component {
     );
   }
 
+  renderBindingStatus() {
+    // check if the service is UPS.
+    // binding status is not shown for other services in the BoundServiceRow view.
+    // binding status is normally shown in UnboundServiceRow views.
+    if(!this.isUPSService()) {
+        return null;
+    }
+
+    return (
+        <ListView.InfoItem key="bind-status">
+            {this.props.service.isBindingOperationInProgress() && (
+                <React.Fragment>
+                    <Icon name="spinner" spin size="lg" />
+                    {this.props.service.getBindingOperation()} In Progress
+                </React.Fragment>
+            )}
+            {this.props.service.isBindingOperationFailed() && (
+                <React.Fragment>
+                    <Icon type="pf" name="error-circle-o" />
+                    Operation Failed. Please Try Again Later.
+                </React.Fragment>
+            )}
+        </ListView.InfoItem>
+    );
+  }
+
+  renderBindingButtons() {
+    // check if the service is UPS. we only allow multiple bindings in case of UPS
```

Comment:
Can this  be externalized  since it is shared between bound an dunbound service rows now?

Reply:
I think this might be too small to extract a new component, but yeah, it doesn't hurt :)

Reply:
Done